nautobot_dns_models/template_content.py

```diff
@@ -0,0 +1,136 @@
+"""Extensions of baseline Nautobot views."""
+
+from urllib.parse import urlencode
+
+from constance import config as constance_config
+from nautobot.apps.ui import ObjectsTablePanel, SectionChoices, TemplateExtension
+from nautobot.core.views.utils import get_obj_from_context
+from netutils.ip import ipaddress_address
+
+from nautobot_dns_models.models import (
+    AAAARecordModel,
+    ARecordModel,
+    DNSZoneModel,
+    PTRRecordModel,
+)
+from nautobot_dns_models.tables import (
+    AAAARecordModelTable,
+    ARecordModelTable,
+    PTRRecordModelTable,
+)
+
+
+class ForwardDNSRecordsTablePanel(ObjectsTablePanel):
+    """Add A/AAAA DNS Records to the right side of the IP Address page."""
+
+    def should_render(self, context):
+        """Check if the table should be rendered."""
+        show_panel = constance_config.nautobot_dns_models__SHOW_FORWARD_PANEL
+        if show_panel == "never":
+            return False
+        if show_panel == "if_present":
+            ip_address = get_obj_from_context(context)
+            if ip_address.ip_version == 4:
+                return ARecordModel.objects.filter(address=ip_address).exists()
+            if ip_address.ip_version == 6:
+                return AAAARecordModel.objects.filter(address=ip_address).exists()
+        return True
+
+    def get_extra_context(self, context):
+        """Set the table class based on the IP version of the IP address."""
+        # Get the IP address from the context.
+        ip_address = get_obj_from_context(context)
+
+        # Use ARecordModelTable for IPv4 and AAAARecordModelTable for IPv6.
+        if ip_address.ip_version == 4:
+            self.table_class = ARecordModelTable
+        elif ip_address.ip_version == 6:
+            self.table_class = AAAARecordModelTable
+
+        # Construct the URL query to auto-populate fields when adding a new record.
+        autopop_fields = {"address": ip_address.id}
+        name, zone = ip_address.dns_name.split(".", 1)
+        if zone in DNSZoneModel.objects.values_list("name", flat=True):
```

Comment:
Might be more efficient to do `if DNSZoneModel.objects.filter(name=zone).exists()`?

Reply:
Replaced.